minecode/management/commands/update_maven_package_data.py

```diff
@@ -0,0 +1,391 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# purldb is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/purldb for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+from dateutil.parser import parse as dateutil_parse
+from os.path import basename
+import copy
+import logging
+import sys
+import traceback
+
+from django.db import transaction
+from django.db.utils import DataError
+from django.utils import timezone
+
+from minecode.models import ProcessingError
+from minecode.management.commands import VerboseCommand
+from packagedb.models import Package
+from packageurl import normalize_qualifiers
+from minecode.collectors.maven import MavenNexusCollector
+
+DEFAULT_TIMEOUT = 30
+
+TRACE = False
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(stream=sys.stdout)
+logger.setLevel(logging.INFO)
+
+
+# This is from https://stackoverflow.com/questions/4856882/limiting-memory-use-in-a-large-django-queryset/5188179#5188179
+class MemorySavingQuerysetIterator(object):
```

Comment:
Is this being used here in this command?
Also maybe we want to have some sort of utils and have this class there as this can be used else where too, like it's used in `packagedb/management/commands/fix_purl_values.py`

Reply:
I've moved this to `minecode/utils.py`. This is only use in `fix_purl_values.py` for now, but I think this is a useful piece of code when we run into memory problems.